kpi/serializers/v2/open_rosa.py

```diff
@@ -0,0 +1,99 @@
+# coding: utf-8
+from rest_framework import serializers
+
+from kpi.interfaces.open_rosa import (
+    OpenRosaFormListInterface,
+    OpenRosaManifestInterface,
+)
+
+
+class FormListSerializer(serializers.Serializer):
+    """
+    This serializer is model-agnostic. The list of objects passed to the
+    serializer must inherit from `kpi.interfaces.open_rosa.OpenRosaFormListInterface`  # noqa
+    to be sure expected methods and properties are defined.
+    """
+    # The PEP-8 naming convention is broken on purpose
+    # Open Rosa XML uses CamelCase
+    # See https://docs.getodk.org/openrosa-form-list/#successful-response-document  # noqa
+    formID = serializers.SerializerMethodField('get_form_id')
+    name = serializers.SerializerMethodField('get_name')
+    hash = serializers.SerializerMethodField('get_name')
+    descriptionText = serializers.SerializerMethodField('get_description')
+    downloadUrl = serializers.SerializerMethodField('get_download_url')
+    manifestUrl = serializers.SerializerMethodField('get_manifest_url')
+
+    def get_description(self, obj):
+        self.__validate_object_inheritance(obj)
+        return obj.descriptionText
+
+    def get_download_url(self, obj):
+        self.__validate_object_inheritance(obj)
+        request = self.context['request']
+        return obj.get_download_url(request)
+
+    def get_form_id(self, obj):
+        self.__validate_object_inheritance(obj)
+        return obj.form_id
+
+    def get_hash(self, obj):
+        self.__validate_object_inheritance(obj)
+        return obj.hash
+
+    def get_manifest_url(self, obj):
+        self.__validate_object_inheritance(obj)
+        request = self.context['request']
+        return obj.get_manifest_url(request)
+
+    def get_name(self, obj):
+        self.__validate_object_inheritance(obj)
+        return obj.name
+
+    def __validate_object_inheritance(
+        self, obj: OpenRosaFormListInterface
+    ) -> bool:
+        """
+        Validates if object inherits from `AbstractOpenRosaFormListModel`.
+        It helps to catch upstream missing properties and methods when
+        rendering data
+        """
+        # Use private variable to test validation only once per instantiation
+        assert issubclass(obj.__class__, OpenRosaFormListInterface)
```

Comment:
how about `isinstance()`
```
In [1]: class AbstractBlahBlah: 
   ...:     pass 
   ...:                                                                                                                              

In [2]: class ConcreteYaddaYadda(AbstractBlahBlah): 
   ...:     pass 
   ...:                                                                                                                              

In [3]: o = ConcreteYaddaYadda()                                                                                                     

In [4]: issubclass(o.__class__, AbstractBlahBlah)                                                                                    
Out[4]: True

In [5]: isinstance(o, AbstractBlahBlah)                                                                                              
Out[5]: True
```

Reply:
Replaced

Reply:
if you really need to verify that the class is a descendant (and not the same), then my suggestion of using `isinstance()` doesn't help. it was clearer the way you had it before, with `issubclass()`.

note that we have #3134 to discuss other kinds of abstract class safeguards.

also, you noticed there was an outdated "private variable" comment.

Reply:
> it was clearer the way you had it before, with `issubclass()`.

but you still need the check to make sure the classes aren't the same, because `issubclass(A, A)` will happily return `True`